Add moneypuck and dbt project

bbrewington opened the pull request:
Changes:
- Python scripts to ingest CSV's & Zip's from "Data" page of [Moneypuck.com](https://moneypuck.com/data.htm) (to be run manually...will automate later)
- [dbt](https://www.getdbt.com/) project for data transformation

Fixes #3

Doing a test on this PR w/ Qodo (AI-based PR slash commands / actions) - not 100% sure I'd just copy+paste the results, and in a PR desc, I might note to refer to the AI-generated comment.  See: https://github.com/bbrewington/nhl-data/pull/2#issuecomment-2945116782

bbrewington committed:
update get_data

bbrewington committed:
add dbt project

bbrewington committed:
add dlt hello world

bbrewington commented:
hyperlint review

bbrewington committed:
update moneypuck

bbrewington committed:
update dbt

bbrewington committed:
update nhl_api

bbrewington commented:
@CodiumAI-Agent /describe

QodoAI-Agent commented:
## Title

#### (Describe updated until commit https://github.com/bbrewington/nhl-data/commit/57b3610bfa8590c74090c26b5fba92bca7e7ebee)


Add moneypuck and dbt project

___
### **User description**
Changes:
- Python scripts to ingest CSV's & Zip's from "Data" page of Moneypuck.com (to be run manually...will automate later)
- dbt project for data transformation


___

### **PR Type**
Enhancement


___

### **Description**
- Add NHL API client, session with retry.

- Script to collect and save NHL API responses.

- Fetch and convert MoneyPuck CSVs to parquet.

- Add S3 upload utility and DBT staging model.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>18 files</summary><table>
<tr>
  <td><strong>collect_responses.py</strong><dd><code>Script to collect and save NHL API JSON responses</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-8bb642a9f219c5e775a00b8002576e693c9a8a125f406c3e3f616e87bdd46625">+242/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_moneypuck_data.py</strong><dd><code>Script to fetch MoneyPuck stats and convert to parquet</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-8394c5769afe31f228d5debe8943822752287f212ddf43c37db63e833c20e565">+225/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>proj_lineup.py</strong><dd><code>Add projected lineup HTML parsing functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-f861b9b442385acd17abdf28cdb7162666f71623aa4f672787036df3b1e90dc2">+123/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>teams.py</strong><dd><code>Implement `TeamsResource` endpoints and validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-c6156b9a2e596b74725fbe080fc827a9b5c286d425072b15b23a72c764159b7d">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>players.py</strong><dd><code>Implement `PlayersResource` methods with validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-91539bb02d9c06647b63d7641d9c3d641dc32b09d70e2804e8908ac057236526">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>write_moneypuck_to_s3.py</strong><dd><code>Script to upload MoneyPuck parquet files to S3</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-4fa00015cffb952e7b2d36885b6211291fa5fa46c779fc32c0e0239d971407ac">+48/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>session.py</strong><dd><code>Add `NHLSession` with backoff and error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-077365dc4aac8de337c35751f055c63aedfd606b95621d11f2591b793e346831">+51/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>league.py</strong><dd><code>Implement `LeagueResource` endpoints with checks</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-a2f7a3dd4db3e79c03d12101a7c43ee41789e0c630e6d0c12aa1d4323e4e3452">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>moneypuck.py</strong><dd><code>Add `get_moneypuck_data` fetcher and saver</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-df086a38065845c9784304de223422190c9d108728f491d5c21203aa1f89cef1">+41/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>aws.py</strong><dd><code>Add AWS S3 upload utility functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-e0b3b7244df5d05c3e8020388bad4d470d2f82c9d38dbf132f6ce45476f77d10">+32/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>games.py</strong><dd><code>Implement `GamesResource` data endpoints</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-eb0a8d36ffa768bfb320d980320f41433078fd56abeced551ff20f11849e3b3f">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>client.py</strong><dd><code>Add `NHLClient` facade for resource access</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-33604c5973927778d0f2c4a0e17da859bed8e88be76719b6097a9b02c459c2c6">+27/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>get_nhl_api_data.py</strong><dd><code>Example script demonstrating `NHLClient` usage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-7d4f06caf6f9401d15d4bd5115ebb033487bf7d6cec5a937b3224b4adf399298">+24/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>base.py</strong><dd><code>Add `BaseResource` wrapper for requests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-6dc16d45892a774ce719ce97716853321ab1cd21166d22c5534b34f9938d62a8">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>player_8478402_mcdavid_game_log_20232024_regular_season.json</strong><dd><code>Add example response for McDavid game log</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-da2b54104a058b47a554e11ce2a53c2166ab34cd056895cf22a922adf6a1348d">+2139/-0</a></td>

</tr>

<tr>
  <td><strong>season_ids.json</strong><dd><code>Add example response for season IDs endpoint</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-cba7cfdd02aad094ebb92441d4af5263ba9b020a8c6081d0add1c147ff6c2e24">+119/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>games_list.json</strong><dd><code>Add example response for all games list</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-f0c868a0d905b3b9c532ba7f973814598fb3691e36474deb3d8c01e76ae85f9f">+60/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__season_segment_goalie_situation.sql</strong><dd><code>Add DBT staging model for goalie situations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-204a4c33f820ac95eabf58fe33becb4182c49272e78e51d11b4eb8911b69d649">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>settings.py</strong><dd><code>Configure project directory and data paths</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-c52387ecbbb2fcb8c331725d66bff428f0a327f1d9802afc21d448b41a825dc3">+27/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>69 files</summary><table>
<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+90/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README_old.md</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-02721b2d7d704bedd62f66db7e8d018e17756d98c1b22b013d43730f22be54a4">+31/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>.env.example</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-c64c802d5b18c4f99a77b9de54c040ca0cd3ca991768c1777991c94c8a46247d">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-f1ff747eca6be41f4dc4894fac678232f77d0ef1ea99e32b44bfc5cdf9e1b3b5">+35/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>dbt_project.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b76d20c5098ce4259769a1862be1190b990aa8d825faf222e2755f9bb90cd576">+23/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>game_id_attrs.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-86538d1a14c1a2dccfadae3c6f4516f82db41bd59d1838446cdff7a1d6d9c760">+19/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>_models.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-d0c2be5a28a49071e68df801096d9506b1d1b6ff44bc07ce8650a441d696bcd4">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>int__players.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b0aea7ad8d3435775f2fa93f3c270c13909e6ac603eefb7f62399efda4eca82e">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>int__team_game.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-713c043bf222cfb366dc828ab3dbed01233d771e3ae9c8a0c761f905d3d032e4">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>_sources.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-81ad9e76de255772efad0c25f430e82c337674f12f57c51b1e887b5b19e579f3">+17/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>_models.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-65d2be8cd34efaf7e688852d9ba09bb92fbc3c09214ca6b492200801d3be3976">+11/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__players_hist.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-0ab0cace3312665634b9d7173497ce05d248062e01cd2ecf061a62db5253aee1">+25/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__players.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-1f5fa665c854cd9d78be3b29df00fb26e3b1b95d019a1d5477e7ab2b31ba3ec1">+10/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__season_segment_line_situation.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-536b72ff8b72b8242fea8d08f3f6202394c4362d709838507aea9846d6f79082">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__season_segment_skater_situation.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b5ba17e2a664b63d2d83c5a91ba8fcda61f037dd869d9f65bd05db3f61e47174">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__shots.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-cee754588ab57cc1d769c4f0712a25f1d9aeb835cdb40b954365e781ec05f16c">+127/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stg_moneypuck__team_game_situation.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b0f3ccd0c8b0fce13182d19807aad724c293f148cbd015b19f221847e59cc9f5">+128/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stg__team_abbrs.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-1f7ef5ec01c670fd6b9937239262815cf39153c2dd45cc6575db62c2b7fcab31">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>stg__teams.sql</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-65292c85750c66974fa8dd02f57ed51329fdbc3576c506be3d790a867d202944">+24/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>package-lock.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-6b1df2cfc2d10cd40b78582bf4b5180c6600267f3cfcf3f42418d1718dd0a0a2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>packages.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-6a66251352f64131bbd7d64b013128ce80bac3548cca013c7689b25581f86b8a">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pyproject.toml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-e7c85c205f45f0523260cc8854d7a8be4db91b9c41f804ede163d2bbeb202742">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>_seeds.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-cd547891c1946035c31939a30d5ba3ea15fc7c96569983bf0491b4f2b7914f02">+22/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>_sources.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-49681d6407e484cb550fbd7150eaa61fc529122c522398ab79768a282e80c593">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>snapshot__all_players_lookup.yml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-ba051c8ad1401a20d7d04e84f9ceeedb472c8793d8aa75127b06cea093cc3965">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>erd.md</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b237a9d626957c2f594b4df413f34836f97ad0af6fdd300ce53aa10c58c0786d">+39/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>.python-version</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-aee57f0e7a487af9b67f1a671d4a305b779c81024fd17e24a96fd5fe56c3f6fa">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pyproject.toml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-46067b8b3886cb40ddff3ecc1ce738b7a94bc375c0d97f4b0ccd344ef9fd8ee7">+14/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-038bea8a6e1ce6995e3adcdf89277107f3a0fbefd925800f77cff2baba609b95">+48/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>.python-version</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-520406f44543a1aac670759ff68378cf31f492c7696c67c1a5135e1b146aefac">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-ce36eb66695663e9e1529d7b45b803271389e490f6209062f0d33b616b75c60b">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pyproject.toml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-e6e97ac0be1dea44267ca2ff9e492101e4c0441ec7ae18cf4b783cf90e5bf26a">+19/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>game.py</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-33b49385f7ce5273740e6a716045f5b7f58082f84a54ae9946d580c9ed4dcf82">+0/-122</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>nhl.py</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-63996c965f27c49b458e869120f83816112729e720bd31981249e5af5504810b">+0/-161</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>player.py</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-7387e5c0c17a58b192bfa46225b12ab274b4d2400889dd0b424dcb648c7ed3c3">+0/-93</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>team.py</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-a5e71155d206558cbb3616986a9df5135b24946bc06025f3fe854a9e979717cb">+0/-206</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>pyproject.toml</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-1a45bd938160f7761769e7b6fbc819d1cc7f84ce37f1a6a23ef83cc4e214abc8">+25/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>__init__.py</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-e8f2fd1c1594556c8281b6e7be2a56e3ccc1c3d1075cf5220e1db4983fac6381">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>dir_file_url_config.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-07bf207e361aa549759381ef4d9e13a20b8d9856f71e39286662bddb6fbfd805">+32/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>game_2023020204_landing.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-dd4ea1bdeceeeb11ee2b610a9128cb1e51ce9813d50c42af4074eada6524c211">+617/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>game_2023020204_play_by_play.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-845ada861dd4a7bb353d515115f6459b6f0cb3f37c7e39e455dac7a0b55ca864">+8068/-0</a></td>

</tr>

<tr>
  <td><strong>game_boxscore.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-3dc52793f40f82f0a19e98738b7aa2c1dcd8c00de340b5f25059f22272bccc08">+1008/-0</a></td>

</tr>

<tr>
  <td><strong>game_info.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-7065163c7945ada594399778830953f83abffae3d330a298acf719752980ef7e">+36/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>game_landing.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-03217e02d6be606ec1ca56855b1f2f766efa18405f4ec60313a5dc8d8e59e366">+617/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>game_play_by_play.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-f8e339c274dd069265984bd02ae02bbfa5ffa642cf3c7823081b014de41cbf85">+8068/-0</a></td>

</tr>

<tr>
  <td><strong>game_shift_charts.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-30cded3143d22226b36b682ad7dd0d2d05d0794d3caf228d7ca7dd806432f38d">+14589/-0</a></td>

</tr>

<tr>
  <td><strong>glossary.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-b2942578c1c447ede90980b022e5f93a7476174cef1616c53b43d30d925acfc8">+2877/-0</a></td>

</tr>

<tr>
  <td><strong>player_8478402_mcdavid.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-8881e64e42c37972cfb7cd43486a6e7c8845c11c280f2c4e5ff860b6ff1090be">+1695/-0</a></td>

</tr>

<tr>
  <td><strong>player_8478402_mcdavid_game_log_now.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-94c2ca878c2842c8d2105c1e898480adbeee9c4f4880541bc36ea9823b364af8">+121/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>player_game_log_current.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-fbfeb2b8301a02cc61c1cefeadbbcb8c1adea831376ca5571d4f8f07391d8378">+121/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>player_game_log_mcdavid.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-e0b57fa67582eb49cb27b5c5bc8d4683adc0c1f98fe97f08a20b95123e9557b4">+2139/-0</a></td>

</tr>

<tr>
  <td><strong>player_mcdavid.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-fb5b7f72a7ac027370494770ff9cd4144aa148977e444e6caccf106978a6155e">+1695/-0</a></td>

</tr>

<tr>
  <td><strong>player_spotlight.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-c36ffd561cd68c050898d81ab246e53d6ff9117d3403598dad83f37e03561895">+157/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>schedule_calendar.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-11e31b68360a0b32d1196a1a3ce72fcf08fcb24148f493a8c2912a6873ed16b3">+727/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>standings_date.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-5268ae52060d3fe5e0bd83172530e7d19e211ca89b48f6b016fa05bd8e3c3a1b">+2931/-0</a></td>

</tr>

<tr>
  <td><strong>standings_now.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-dd8a97476ac09eba2c4dbb7684e4851e3df961b29000686599ae81c2e8fbd789">+2965/-0</a></td>

</tr>

<tr>
  <td><strong>stats_leaders_goalie_current.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-1b952a7e207e928da0754db830dcf6dbfd9d3e3151fe02231779bdfd63404e64">+405/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stats_leaders_goals_limited.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-dfc93244874a8dd87970b4f68b5a9920fec9834669f9521bdd16c92f6cf41341">+199/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stats_leaders_skater_20232024_regular.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-5a39da6473bbda44ca62ec624fc7b4c6c16ebb19bbeb9b1fd2010fb61b132361">+864/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stats_leaders_skater_2023_regular.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-bb839af62c0a9cebe77899ee3ece15afc69885cc4b81217a5ebb832f40518cce">+864/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stats_leaders_skater_current.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-3adb7054aba0d6be0e0dc8dd435caef79080b7593ee9970f1158a44e1b28d3b1">+848/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>stats_leaders_skater_goals_limit10.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-e845a4ada2ffe3f3ebf128b2c72ac28dc9515e0612d5ceb9d09a74c9de7a2a22">+199/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>team_info_list.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-bc65c6a25e11a14074be9edf5ef8cae07f6389e5b2ae21db78f87f67ecf829b2">+511/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>team_schedule_stl.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-99efc4427ec96ca5d5dca373f7856ae77dc860502caca5dfb9a601a96eff64e1">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>team_scoreboard_stl.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-f4c6230b103f35db48f0602d9c979277405bea8a3374fb33cf10f294b0fedb44">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>team_stats_stl_now.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-33e23f43bfb8fbdc75cd774b3e8153d0a5d1781ef6abda8a5f7517efc9c77f25">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>team_stats_stl_season.json</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-df6adb0ab908817306c756bef22caa69a47801e6b75582497e1440b7b29ade06">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>__init__.py</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-658f33c3803c282d9721b0296207fce403ac583f99d4c8f37be9908b55fda203">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>requirements.txt</strong></td>
  <td><a href="https://github.com/bbrewington/nhl-data/pull/2/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

bbrewington committed:
add ERD, update API example responses

bbrewington commented:
/update_changelog

bbrewington commented:
/update_changelog --pr_update_changelog.push_changelog_changes=true

github-actions[bot] committed:
[skip ci] Update CHANGELOG.md

bbrewington committed:
Update dbt models, upgrade dbt-duckdb/duckdb, add project docs

  - Expand column renaming in staging models (shots, skater/goalie/line                                                          situations) to follow snake_case conventions with MoneyPuck prefix
    mappings (xGoals -> exp_goals, I_F_ -> ind_for_, etc.)
  - Add int__players model joining snapshot history with current player data
  - Refactor int__team_game to join through stg__team_abbrs for normalized
    team abbreviations
  - Add S3 external sources definition in staging/_models.yml
  - Upgrade dbt-duckdb to 1.10.x and pin duckdb to 1.4.x (resolves
    MotherDuck deprecation warning)
  - Add profiles.yml with dev (local DuckDB) and prod (MotherDuck) targets
  - Add CLAUDE.md, README, TODO, DAG diagram, and .gitignore updates
  - Remove stale hockey_stats_links seed

bbrewington committed:
Rename project to nhl_dbt, fix MoneyPuck scraper, upgrade dbt-duckdb

  - Rename dbt project/profile from moneypuck_dbt to nhl_dbt                                                          - Add Cloudflare cf_clearance cookie support to MoneyPuck data fetcher
    (--cf-clearance flag) and allow local HTML fallback (--local-html)
  - Fix config path resolution in get_moneypuck_data.py to use
    src/moneypuck/config/ instead of relative paths
  - Use browser-like headers for moneypuck.com requests
  - Update shot data URLs to current archives (shots_2007-2024, shots_2025)
  - Add s3_region setting to dev profile for local DuckDB S3 access
  - Standardize env var names to AWS_ACCESS_KEY_ID/AWS_SECRET_ACCESS_KEY

bbrewington committed:
Rename python/ to ingestion/, fix dbt models, clean up get_data/

  dbt improvements:
  - Fix penality typo: rename to penalty_minutes_for/against
  - Resolve opposing_team_abbr in int__team_game via stg__team_abbrs
  - Add uniqueness tests to season segment models (skater/goalie/line)
  - Add composite unique test on shots (season + game_id + shot_id)
  - Add column-level descriptions to all _models.yml files

  Ingestion improvements:
  - Add Cloudflare cf_clearance cookie support for MoneyPuck downloads
  - Add --local-html flag for offline HTML parsing
  - Fix config path resolution to use src/moneypuck/config/
  - Update shot data URLs to current archives (2007-2024, 2025)

  Repo cleanup:
  - Rename python/ to ingestion/
  - Move READMEs from get_data/ to ingestion/src/ and delete get_data/
  - Update CLAUDE.md and README.md to reflect new folder names
